src/shared/tpm2-util.c

```diff
@@ -630,9 +646,30 @@ static int tpm2_make_encryption_session(
                         TPMA_SESSION_CONTINUESESSION;
         ESYS_TR session = ESYS_TR_NONE;
         TSS2_RC rc;
-
+        TPM2B_AUTH auth = { 0 };
```

Comment:
also, move to innermost scope, i.e. inside the if block?

Reply:
> also, move to innermost scope, i.e. inside the if block?

I was wondering that, I looked around at other code and everything seemed to declare at the entry of the function. I wasn't sure if you're trying to be C89 compatible.